lib/cc/engine/analyzers/engine_config.rb

```diff
@@ -0,0 +1,48 @@
+module CC
+  module Engine
+    module Analyzers
+      class EngineConfig
+        attr_reader :config
+
+        def initialize(hash)
+          normalize(hash)
+        end
+
+        def language(language)
+          config.
+            fetch("languages", {}).
+            fetch(language, {})
+        end
+
+        def languages
+          config.fetch("languages", {})
+        end
+
+        private
+
+        attr_reader :hash
+
+        def normalize(hash)
+          @config = hash.fetch("config", {})
+          @config["languages"] = build_language_config
+        end
+
+        def build_language_config
+          languages = config.fetch("languages", {})
+
+          if languages.is_a?(Array)
+            languages.each_with_object({}) do |language, map|
+              map[language.downcase] = {}
+            end
+          elsif languages.is_a?(Hash)
+            languages.each_with_object({}) do |(key, value), map|
+              map[key.downcase] = value
+            end
+          else
+            hash.languages = {}
```

Comment:
@BlakeWilliams where does `hash` get assigned? I see it's an `attr_reader`, but it's unclear where it's actually set.